wasm-pack.yaml

```
@@ -0,0 +1,151 @@
+package:
+  name: wasm-pack
+  version: 0.12.1
+  epoch: 0
+  description: rust to wasm build tool
+  copyright:
+    - license: Apache-2.0
+
+environment:
+  contents:
+    packages:
+      - autoconf
+      - automake
+      - build-base
+      - busybox
+      - bzip2-dev
+      - ca-certificates-bundle
+      - cargo-auditable
+      - curl-dev
+      - openssl-dev
+      - rust
+      - zstd-dev
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/rustwasm/wasm-pack
+      expected-commit: e25150ecd322fc03c9abcd141bc979b108e8a344
+      tag: v${{package.version}}
+
+  - uses: patch
+    with:
+      patches: upgrade-deps.patch
+
+  - runs: |
+      # prepare
+      rm -rf Cargo.lock
+      cargo fetch --target="${{host.triplet.rust}}"
+
+      cargo auditable build --release --frozen
+      install -Dm755 target/release/wasm-pack \
+      -t "${{targets.contextdir}}"/usr/bin
+
+  - uses: strip
+
+update:
+  enabled: true
+  github:
+    identifier: rustwasm/wasm-pack
+    strip-prefix: v
+
+test:
+  environment:
+    contents:
+      packages:
+        - nodejs
+        - npm
+        - rustup
+        - yarn
+        - build-base
+        - binaryen
+    environment:
+      HOME: /home/build
+  pipeline:
+    - runs: |
+        # Function to display messages
+        function print_info {
+            echo -e "\033[1;34m[INFO]\033[0m $1"
+        }
+
+        function print_error {
+            echo -e "\033[1;31m[ERROR]\033[0m $1"
+            exit 1
+        }
+
+        print_info "Installing default Rust toolchain via Rustup..."
+        rustup install stable || print_error "Failed to install Rust via Rustup"
+        rustup default stable || print_error "Failed to set default Rust toolchain"
+        ARCH=$(uname -m)
+        export PATH="$HOME/.rustup/toolchains/stable-${ARCH}-unknown-linux-gnu/bin:$PATH"
+        cargo version
+
+        # Ensure the wasm32-unknown-unknown target is installed
+        print_info "Checking for wasm32-unknown-unknown target..."
+        set +e
+        rustup target list --installed | grep wasm32-unknown-unknown &> /dev/null
+        if [ $? -ne 0 ]; then
+            print_info "Installing wasm32-unknown-unknown target..."
+            rustup target add wasm32-unknown-unknown || print_error "Failed to install wasm32-unknown-unknown target."
+        else
+            print_info "wasm32-unknown-unknown target is already installed."
+        fi
```

```suggestion
        rustup target add wasm32-unknown-unknown || print_error "Failed to install wasm32-unknown-unknown target."
```

This will succeed if it's already installed (which it shouldn't be anyway)